### PR TITLE
PR - Fargate OSS pattern logs enabled | FluentBit

### PR DESCRIPTION
**Pull request by Howlla**
*Issue #, if available:* 

*Description of changes:* Remove hardcoded coreDNS version to "auto" and updated fargate cluster provider to kube 1.28

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/single-new-eks-awsnative-fargate-observability-construct/index.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default class SingleNewEksAWSNativeFargateobservabilityConstruct {
         // Define fargate cluster provider and pass the profile options
         const fargateClusterProvider : blueprints.FargateClusterProvider = new blueprints.FargateClusterProvider({
             fargateProfiles,
-            version: eks.KubernetesVersion.of("1.27")
+            version: eks.KubernetesVersion.of("1.28")
```
Comment (elamaran11) [inline lib/single-new-eks-awsnative-fargate-observability-construct/index.ts:42]: Change this to `auto`. We cant do this for each Kubernetes version to EKS.

File: lib/single-new-eks-fargate-opensource-observability-pattern/index.ts [hunk 2]
```diff
@@ -143,7 +143,7 @@ export default class SingleNewEksFargateOpenSourceObservabilityConstruct {
         // Define fargate cluster provider and pass the profile options
         const fargateClusterProvider: blueprints.FargateClusterProvider = new blueprints.FargateClusterProvider({
             fargateProfiles,
-            version: eks.KubernetesVersion.of("1.27")
+            version: eks.KubernetesVersion.of("1.28")
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/index.ts:146]: Change this to `auto`. We cant do this for each Kubernetes version to EKS.

File: lib/single-new-eks-fargate-opensource-observability-pattern/index.ts [hunk 3]
```diff
@@ -98,7 +98,7 @@ export default class SingleNewEksFargateOpenSourceObservabilityConstruct {
         const addOns: Array<blueprints.ClusterAddOn> = [
             new blueprints.addons.VpcCniAddOn(),
             new blueprints.addons.CoreDnsAddOn({
-                version: "v1.10.1-eksbuild.1",
+                version: "auto",
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/index.ts:101]: this is inaccurate. We dont support auto with individual addons. You have to add the latest supported version,

**Review by elamaran11 — COMMENTED** [review]
@Howlla Thanks for the PR. Have minor comments.

**Review by elamaran11 — COMMENTED** [review]
LGTM. Let me run e2e.

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-awsnative-fargate-observability deploy

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-awsnative-fargate-observability deploy

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-awsnative-fargate-observability destroy

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-fargate-opensource-observability destroy

**Comment by elamaran11** [comment]
@Howlla Delete stack is failing with below error. Please fix this

```
single-new-eks-fargate-opensource-observability-accelerator |  53 | 5:35:41 PM | DELETE_FAILED        | AWS::CloudFormation::Stack            | single-new-eks-fargate-opensource-observability-accelerator The following resource(s) failed to delete: [singleneweksfargateopensourceobservabilityacceleratorCoreDnsComputeTypePatchDC03326B].

Failed resources:
single-new-eks-fargate-opensource-observability-accelerator | 5:33:28 PM | DELETE_FAILED        | Custom::AWSCDK-EKS-KubernetesPatch    | single-new-eks-fargate-opensource-observability-accelerator/CoreDnsComputeTypePatch/Resource/Default (singleneweksfargateopensourceobservabilityacceleratorCoreDnsComputeTypePatchDC03326B) Received response status [FAILED] from custom resource. Message returned: Error: b'Error from server (NotFound): deployments.apps "coredns" not found\n'
```

**Review by elamaran11 — COMMENTED** [review]
@Howlla Impressive work. I think the documentation is not update to date. [This](https://aws-observability.github.io/cdk-aws-observability-accelerator/patterns/single-new-eks-observability-accelerators/single-new-eks-fargate-opensource-observability/#viewing-logs) section is incorrect on logging, we need to update this to what you have done with some screenshots.

**Comment by Howlla** [comment]
done @elamaran11.

**Comment by elamaran11** [comment]
@Howlla Doc Markdown failures, pls fix those.

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        Match kube.*
+        Merge_Log On
+        Keep_Log Off
+        Buffer_Size 0
+        Kube_Meta_Cache_TTL 300s
+  output.conf: |
+    [OUTPUT]
+        Name cloudwatch_logs
+        Match   kube.*
+        region us-east-1
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml:32]: Why is the region hardcoded, we should pass it as a parameter via YTPL extension

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        log_group_name fargate-observability
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml:33]: Can we also pass log stream name and group name as inputs too. You can check our AMP Addon for example of using YTPL file.

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import 'source-map-support/register';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import * as eks from "aws-cdk-lib/aws-eks";
+import { Construct } from 'constructs';
+import { loadYaml, readYamlDocument } from '@aws-quickstart/eks-blueprints/dist/utils';
+
+export class FluentBitConfigMap implements blueprints.ClusterAddOn {
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts:7]: Can yu add js doc inline in file on what this addon is doing.?

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+kind: Namespace
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfig.yml:1]: move this file to common folder under a new sub folder

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@Howlla Pls fix these

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Log Group Name in cloudwatch
+     */
+    logGroupName: string
+
+    /**
+     * Prefix for logs stream
+     */
+    log_stream_prefix: string;
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts:24]: Use Came case

File: lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    enableFlbProcessLogs?: boolean
+}
+
+/**
+ * Default props for the add-on.
+ */
+const defaultProps: FluentBitConfigMapProps = {
+    awsRegion: "us-east-1",
+    logGroupName: "fargate-observability",
+    log_stream_prefix: "from-fluent-bit-",
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/fluentbitconfigmap.ts:38]: Use Camelcase

File: lib/single-new-eks-fargate-opensource-observability-pattern/index.ts [hunk 10]
```diff
@@ -111,11 +113,17 @@ export default class SingleNewEksFargateOpenSourceObservabilityConstruct {
             );
         }
 
+        const fluentBitConfigMapProps = {
+            awsRegion: region,
+            logGroupName: "fargate-observability",
+            log_stream_prefix: "from-fluent-bit-",
```
Comment (elamaran11) [inline lib/single-new-eks-fargate-opensource-observability-pattern/index.ts:119]: camel case

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Some convention issues.

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-fargate-opensource-observability deploy

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-fargate-opensource-observability destroy

**Comment by elamaran11** [comment]
Expected Fargate Destroy Failure. Merging the PR.

**Pull request merged**